035-cluster-wide-volume-usage-quota-managemnet.md

```diff
@@ -89,14 +89,116 @@ To make all this work the quota plugin will need to introduce some new interface
 3. A Data Source - To publish the current volume usage for the broker in question.
 
 One can envisage a model where the Quota policy was externalised to a separate service which calculated the quota factor
-and thus the Quota Factor Source would just provider a wrapper round that external service. For the purposes of this
+and thus the Quota Factor Source would just provide a wrapper around that external service. For the purposes of this
 proposal all three interfaces would be implemented with the current quota plugin.
 
 In this proposal the Factor Source implementation would have the kafka consumer and delegate to the quota policy to
 determine what factor to apply.
 
 While the Data source would periodically calculate and publish the usage metrics for the local broker.
 
+##### Proposed interface details
+1. QuotaPolicy
+```java
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.quotas.policy;
+
+import io.strimzi.kafka.quotas.VolumeDetails;
+
+/**
+ * Abstracts the decision-making around hard and soft limits and how to calculate the affect the impact breaching the limits has on the client request.
+ */
+public interface QuotaPolicy {
+
+    /**
+     * Does the particular volume breach this policy's defined hard limit.
+     *
+     * @param volumeDetails details of the disk in question.
+     * @return <code>true</code> if this policy considers the volume to breach the limit otherwise <code>false</code>
+     */
+    boolean breachesHardLimit(VolumeDetails volumeDetails);
```

Comment:
I'm wondering if this should accept  `Set<VolumeDetails>` instead so that global capacity quotas could be enforced, purely for backwards compatibility. 

However I don't know if that backwards compatibility is actually worth having as it's not really protecting the cluster/broker from running out of disk space.